imagebuildah/build.go

```diff
@@ -0,0 +1,334 @@
+			Env:   builder.UpdatedEnv(),
+			Image: from,
+		},
+	}
+	err = ib.FromImage(&dImage, node)
+	if err != nil {
+		if err2 := builder.Delete(); err2 != nil {
+			logrus.Debugf("error deleting container which we failed to update: %v", err2)
+		}
+		return fmt.Errorf("error updating build context: %v", err)
+	}
+	_, err = builder.Mount("")
+	if err != nil {
+		if err2 := builder.Delete(); err2 != nil {
+			logrus.Debugf("error deleting container which we failed to mount: %v", err2)
+		}
+		return fmt.Errorf("error mounting new container: %v", err)
+	}
+	b.builder = builder
+	return nil
+}
+
+func (b *buildahExecutor) Delete() (err error) {
+	if b.builder != nil {
+		err = b.builder.Delete()
+		b.builder = nil
+	}
+	return err
+}
+
+func (b *buildahExecutor) Execute(ib *imagebuilder.Builder, node *parser.Node) error {
+	for i, node := range node.Children {
+		step := ib.Step()
+		if err := step.Resolve(node); err != nil {
+			return fmt.Errorf("error resolving step %+v: %v", *node, err)
+		}
+		logrus.Debugf("Parsed Step: %+v", *step)
+		if !b.quiet {
+			fmt.Fprintf(os.Stderr, "STEP %d: %s\n", b.counter, step.Original)
+		}
+		b.counter++
+		requiresStart := false
+		if i < len(node.Children)-1 {
+			requiresStart = ib.RequiresStart(&parser.Node{Children: node.Children[i+1:]})
+		}
+		err := ib.Run(step, b, requiresStart)
+		if err != nil {
+			return fmt.Errorf("error building at step %+v: %v", *step, err)
+		}
+	}
+	return nil
+}
+
+func (b *buildahExecutor) Commit(ib *imagebuilder.Builder, node *parser.Node) (err error) {
+	var imageRef types.ImageReference
+	if b.output != "" {
+		imageRef, err = alltransports.ParseImageName(b.output)
+		if err != nil {
+			imageRef2, err2 := is.Transport.ParseStoreReference(b.store, b.output)
+			if err2 == nil {
+				imageRef = imageRef2
+				err = nil
+			}
+		}
+	} else {
+		imageRef, err = is.Transport.ParseStoreReference(b.store, "@"+stringid.GenerateRandomID())
+	}
+	if imageRef != nil {
+		logName := transports.ImageName(imageRef)
+		logrus.Debugf("COMMIT %q", logName)
+		if !b.quiet {
+			fmt.Fprintf(os.Stderr, "STEP %d: COMMIT %s\n", b.counter, logName)
+		}
+	} else {
+		logrus.Debugf("COMMIT")
+		if !b.quiet {
+			fmt.Fprintf(os.Stderr, "STEP %d: COMMIT\n", b.counter)
+		}
+	}
+	b.counter++
+	if err != nil {
+		return err
+	}
+	options := buildah.CommitOptions{
+		Compression:         b.compression,
+		SignaturePolicyPath: b.signaturePolicyPath,
+	}
+	return b.builder.Commit(imageRef, options)
+}
+
+func (b *buildahExecutor) Build(ib *imagebuilder.Builder, node *parser.Node) (err error) {
+	if err = b.Prepare(ib, node); err != nil {
+		return err
+	}
+	defer b.Delete()
+	if err = b.Execute(ib, node); err != nil {
+		return err
+	}
+	if err = b.Commit(ib, node); err != nil {
+		return err
+	}
+	return nil
+}
+
+// BuildDockerfile builds an image using instructions from the named file.
+func BuildDockerfileFromFile(store storage.Store, options BuildDockerfileOptions, dockerfile ...string) error {
```

Comment:
If you change the name to BuildDockerfileFromFile in build.go, you probably want to do the same here.